internal/util/rocksmq/server/rocksmq/rocksmq_impl.go

```diff
@@ -882,3 +885,159 @@ func (rmq *rocksmq) updateAckedInfo(topicName, groupName string, ids []UniqueID)
 	}
 	return nil
 }
+
+func (rmq *rocksmq) ReaderSeek(topicName string, msgID UniqueID) {
+	rmq.readerOffset.Store(topicName, msgID)
+}
+
+func (rmq *rocksmq) Next(ctx context.Context, topicName string, messageIDInclusive bool) (ConsumerMessage, error) {
+	ll, ok := topicMu.Load(topicName)
+	if !ok {
+		return ConsumerMessage{}, fmt.Errorf("topic name = %s not exist", topicName)
+	}
+	lock, ok := ll.(*sync.Mutex)
+	if !ok {
+		return ConsumerMessage{}, fmt.Errorf("get mutex failed, topic name = %s", topicName)
+	}
+	lock.Lock()
+	defer lock.Unlock()
+	fixChanName, err := fixChannelName(topicName)
+	if err != nil {
+		log.Debug("RocksMQ: fixChannelName " + topicName + " failed")
+		return ConsumerMessage{}, err
+	}
+	readOpts := gorocksdb.NewDefaultReadOptions()
+	defer readOpts.Destroy()
+
+	var msg ConsumerMessage
+	var msgID UniqueID
+	if id, ok := rmq.readerOffset.Load(topicName); ok {
+		msgID = id.(UniqueID)
+	} else {
+		return ConsumerMessage{}, fmt.Errorf("topic name = %s not exist in readerOffset", topicName)
+	}
+	readOpts.SetPrefixSameAsStart(true)
+	iter := rmq.store.NewIterator(readOpts)
+	defer iter.Close()
+	dataKey := path.Join(fixChanName, strconv.FormatInt(msgID, 10))
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.Debug("Stop get next reader message!")
+			return ConsumerMessage{}, nil
+		default:
+			if iter.Seek([]byte(dataKey)); !iter.Valid() {
+				continue
+			}
```

Comment:
If the `dataKey`  is invalid, the block will be running forever

Reply:
This is by design, when invalid, Next() should block until context done.

Reply:
Spinning here will consume lots of CPU and disk I/O